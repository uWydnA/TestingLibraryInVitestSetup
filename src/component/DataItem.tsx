import { DataItemType } from "./DataList";

type ItemProps = {
  editable?: boolean;
  horizontal?: boolean;
  item: DataItemType;
};

export function DataItem({ item, editable, horizontal }: ItemProps) {
  return (
    <div
      className="dataitem"
      key={item.id}
      style={{ display: horizontal ? "flex" : "block" }}
    >
      <div>
        <span>id:{item.id}</span>
        <span>name:{item.name}</span>
      </div>

      {editable && (
        <div>
          <button>编辑</button>
          <button>删除</button>
        </div>
      )}
    </div>
  );
}
