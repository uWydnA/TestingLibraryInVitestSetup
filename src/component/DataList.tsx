import { DataItem } from "./DataItem";

type Props = {
  editable?: boolean;
  horizontal?: boolean;
  data: DataItemType[];
};
export type DataItemType = { name: string; id: string };

export function DataList({ data, editable, horizontal }: Props) {
  return (
    <div>
      {data.map((item) => {
        return (
          <DataItem
            key={item.id}
            editable={editable}
            item={item}
            horizontal={horizontal}
          />
        );
      })}
    </div>
  );
}
