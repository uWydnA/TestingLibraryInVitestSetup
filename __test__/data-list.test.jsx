import { describe, expect, it } from "vitest";
import { logDOM, render, screen } from "@testing-library/react";
import { DataList } from "../src/component/DataList";
const mockData = new Array(2)
  .fill(true)
  .map((_, index) => ({ name: `${index}`, id: `${index}` }));
describe("DataList", () => {
  it("it should be render", () => {
    const { container } = render(<DataList data={mockData} />);
    expect(container).toMatchSnapshot();
  });

  it("it should be render with data", () => {});

  it.skip("it should be render with className", () => {});

  it("it should be editable with editable props", () => {
    const { container } = render(<DataList data={mockData} editable />);
    expect(screen.queryAllByText("编辑")).toHaveLength(mockData.length);
    expect(screen.queryAllByText("删除")).toHaveLength(mockData.length);
  });

  it("it should not be editable without editable props", () => {
    const { container } = render(<DataList data={mockData} />);
    expect(screen.queryAllByText("编辑")).toHaveLength(0);
    expect(screen.queryAllByText("删除")).toHaveLength(0);
  });

  it("it should be horizontal with horizontal props", () => {
    const { container } = render(<DataList data={mockData} horizontal />);
    const targets = document.querySelectorAll(".dataitem");
    const matchs = [...targets].filter((item) =>
      item.getAttribute("style").includes("flex")
    );
    expect(matchs).toHaveLength(mockData.length);
  });

  it("it should not be horizontal with horizontal props", () => {
    const { container } = render(<DataList data={mockData} />);
    const targets = document.querySelectorAll(".dataitem");
    const matchs = [...targets].filter((item) =>
      item.getAttribute("style").includes("block")
    );
    expect(matchs).toHaveLength(mockData.length);
  });
});
