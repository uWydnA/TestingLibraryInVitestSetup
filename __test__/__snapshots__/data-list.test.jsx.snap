// Vitest Snapshot v1

exports[`DataList > it should be render 1`] = `
<div>
  <div>
    <div
      class="dataitem"
      style="display: block;"
    >
      <div>
        <span>
          id:
          0
        </span>
        <span>
          name:
          0
        </span>
      </div>
    </div>
    <div
      class="dataitem"
      style="display: block;"
    >
      <div>
        <span>
          id:
          1
        </span>
        <span>
          name:
          1
        </span>
      </div>
    </div>
  </div>
</div>
`;
